src/sdk/sdk.ts

```diff
@@ -196,6 +203,10 @@ export class Sdk {
   constructor(walletProvider: WalletProviderLike, optionsLike?: EnvNames | SdkOptions) {
     let options: SdkOptions = {};
 
+    if (isWalletConnectProvider(walletProvider)) {
```

Comment:
this lets to avoid any additional mappings before passing Wallet Connect provider to Etherspot SDK constructor